npm failure on circleci build

e.g. https://app.circleci.com/pipelines/github/cgolubi1/buttonmen/64/workflows/54f02579-b09a-4dd0-9344-8ac92c7c30d9/jobs/317

Failure is in `Run grunt JS lint tests`, and looks like:

```
#!/bin/bash -eo pipefail
sh ./util/grunt/circleci-grunt.sh

+ echo Checking to see if npm is in your path.  If not, install it before proceeding
Checking to see if npm is in your path.  If not, install it before proceeding
+ type npm
npm is /usr/bin/npm
+ echo Using npm to install needed grunt libraries
Using npm to install needed grunt libraries
+ [ -f node_modules/grunt-cli/package.json ]
+ npm install grunt-cli
WARN engine grunt-cli@1.4.2: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine grunt-cli@1.4.2: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine liftup@3.0.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine findup-sync@4.0.0: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine micromatch@4.0.3: wanted: {"node":">=8.6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine braces@3.0.2: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine picomatch@2.2.2: wanted: {"node":">=8.6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine fill-range@7.0.1: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine to-regex-range@5.0.1: wanted: {"node":">=8.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
my-project-name@0.1.0 /root/project/util/grunt
`-- grunt-cli@1.4.2  extraneous

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt/package.json ]
+ npm install grunt
npm WARN deprecated coffee-script@1.3.3: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt@0.4.5 
| +-- async@0.1.22 
| +-- coffee-script@1.3.3 
| +-- colors@0.6.2 
| +-- dateformat@1.0.2-1.2.3 
| +-- eventemitter2@0.4.14 
| +-- exit@0.1.2 
| +-- findup-sync@0.1.3 
| | +-- glob@3.2.11 
| | | +-- inherits@2.0.4 
| | | `-- minimatch@0.3.0 
| | `-- lodash@2.4.2 
| +-- getobject@0.1.0 
| +-- glob@3.1.21 
| | +-- graceful-fs@1.2.3 
| | `-- inherits@1.0.2 
| +-- grunt-legacy-log@0.1.3 
| | +-- grunt-legacy-log-utils@0.1.1 
| | | +-- lodash@2.4.2 
| | | `-- underscore.string@2.3.3 
| | +-- lodash@2.4.2 
| | `-- underscore.string@2.3.3 
| +-- grunt-legacy-util@0.2.0 
| | `-- which@1.0.9 
| +-- hooker@0.2.3 
| +-- iconv-lite@0.2.11 
| +-- js-yaml@2.0.5 
| | +-- argparse@0.1.16 
| | | +-- underscore@1.7.0 
| | | `-- underscore.string@2.4.0 
| | `-- esprima@1.0.4 
| +-- lodash@0.9.2 
| +-- minimatch@0.2.14 
| | +-- lru-cache@2.7.3 
| | `-- sigmund@1.0.1 
| +-- nopt@1.0.10 
| +-- rimraf@2.2.8 
| +-- underscore.string@2.2.1 
| `-- which@1.0.9 
`-- grunt-cli@1.4.2  extraneous

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-jshint/package.json ]
+ npm install grunt-contrib-jshint
WARN engine esprima@4.0.0-dev: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.2  extraneous
+-- grunt-contrib-jshint@0.4.3 
`-- jshint@1.1.0 
  +-- cli@0.4.5 
  +-- esprima@4.0.0-dev 
  +-- peakle@0.0.1 
  +-- shelljs@0.1.4 
  `-- underscore@1.4.4 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-concat/package.json ]
+ npm install grunt-contrib-concat
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.2  extraneous
`-- grunt-contrib-concat@0.3.0 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-uglify/package.json ]
+ npm install grunt-contrib-uglify
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.2  extraneous
`-- grunt-contrib-uglify@0.2.7 
  +-- grunt-lib-contrib@0.6.1 
  | `-- zlib-browserify@0.0.1 
  `-- uglify-js@2.4.24 
    +-- async@0.2.10 
    +-- source-map@0.1.34 
    | `-- amdefine@1.0.1 
    +-- uglify-to-browserify@1.0.2 
    `-- yargs@3.5.4 
      +-- camelcase@1.2.1 
      +-- decamelize@1.2.0 
      +-- window-size@0.1.0 
      `-- wordwrap@0.0.2 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-lint-inline/package.json ]
+ npm install grunt-lint-inline
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.2  extraneous
`-- grunt-lint-inline@0.1.1 
  `-- grunt-contrib-jshint@0.3.0 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
npm WARN deprecated nodeunit@0.7.4: you are strongly encouraged to use other testing options
WARN engine tap@15.0.2: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine which@2.0.2: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine istanbul-lib-processinfo@2.0.2: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tap-mocha-reporter@5.0.1: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine libtap@1.1.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine coveralls@3.1.0: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine jackspeak@1.4.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine foreground-child@2.0.0: wanted: {"node":">=8.0.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine mkdirp@1.0.4: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine chokidar@3.5.1: wanted: {"node":">= 8.10.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tcompare@5.0.4: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine nyc@15.1.0: wanted: {"node":">=8.9"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine anymatch@3.1.2: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine glob-parent@5.1.2: wanted: {"node":">= 6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine is-binary-path@2.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine fsevents@2.3.2: wanted: {"node":"^8.16.0 || ^10.6.0 || >=11.0.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine readdirp@3.5.0: wanted: {"node":">=8.10.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine binary-extensions@2.2.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine request@2.88.2: wanted: {"node":">= 6"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
WARN engine har-validator@5.1.5: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
WARN engine punycode@2.1.1: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine cross-spawn@7.0.3: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine path-key@3.1.1: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine shebang-command@2.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine shebang-regex@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine p-map@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine istanbul-lib-coverage@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine make-dir@3.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine aggregate-error@3.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine clean-stack@2.2.0: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine indent-string@4.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine minipass@3.1.3: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine async-hook-domain@2.0.2: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine bind-obj-methods@3.0.0: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tap-parser@10.1.0: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine stack-utils@2.0.3: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine escape-string-regexp@2.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm ERR! Linux 4.15.0-1092-aws
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
npm ERR! typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
npm ERR! typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
npm ERR! typeerror     at /usr/share/npm/lib/install/deps.js:457:7
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
npm ERR! typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
npm ERR! typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
npm ERR! typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /root/project/util/grunt/npm-debug.log
/root/project/util/grunt/node_modules/micromatch/index.js:44
    let isMatch = picomatch(String(patterns[i]), { ...options, onResult }, true);
                                                   ^^^

SyntaxError: Unexpected token ...
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/root/project/util/grunt/node_modules/findup-sync/index.js:12:10)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)

Exited with code exit status 1

CircleCI received exit code 1
```

Part of the error is a red herring.  Here's the output from my last successful run, a week ago:

```
#!/bin/bash -eo pipefail
sh ./util/grunt/circleci-grunt.sh

+ echo Checking to see if npm is in your path.  If not, install it before proceeding
Checking to see if npm is in your path.  If not, install it before proceeding
+ type npm
npm is /usr/bin/npm
+ echo Using npm to install needed grunt libraries
Using npm to install needed grunt libraries
+ [ -f node_modules/grunt-cli/package.json ]
+ npm install grunt-cli
WARN engine grunt-cli@1.4.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine grunt-cli@1.4.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
WARN engine atob@2.1.2: wanted: {"node":">= 4.5.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
my-project-name@0.1.0 /root/project/util/grunt
`-- grunt-cli@1.4.1  extraneous

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt/package.json ]
+ npm install grunt
npm WARN deprecated coffee-script@1.3.3: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt@0.4.5 
| +-- async@0.1.22 
| +-- coffee-script@1.3.3 
| +-- colors@0.6.2 
| +-- dateformat@1.0.2-1.2.3 
| +-- eventemitter2@0.4.14 
| +-- exit@0.1.2 
| +-- findup-sync@0.1.3 
| | +-- glob@3.2.11 
| | | +-- inherits@2.0.4 
| | | `-- minimatch@0.3.0 
| | `-- lodash@2.4.2 
| +-- getobject@0.1.0 
| +-- glob@3.1.21 
| | +-- graceful-fs@1.2.3 
| | `-- inherits@1.0.2 
| +-- grunt-legacy-log@0.1.3 
| | +-- grunt-legacy-log-utils@0.1.1 
| | | +-- lodash@2.4.2 
| | | `-- underscore.string@2.3.3 
| | +-- lodash@2.4.2 
| | `-- underscore.string@2.3.3 
| +-- grunt-legacy-util@0.2.0 
| | `-- which@1.0.9 
| +-- hooker@0.2.3 
| +-- iconv-lite@0.2.11 
| +-- js-yaml@2.0.5 
| | +-- argparse@0.1.16 
| | | +-- underscore@1.7.0 
| | | `-- underscore.string@2.4.0 
| | `-- esprima@1.0.4 
| +-- lodash@0.9.2 
| +-- minimatch@0.2.14 
| | +-- lru-cache@2.7.3 
| | `-- sigmund@1.0.1 
| +-- nopt@1.0.10 
| +-- rimraf@2.2.8 
| +-- underscore.string@2.2.1 
| `-- which@1.0.9 
`-- grunt-cli@1.4.1  extraneous

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-jshint/package.json ]
+ npm install grunt-contrib-jshint
WARN engine esprima@4.0.0-dev: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.1  extraneous
+-- grunt-contrib-jshint@0.4.3 
`-- jshint@1.1.0 
  +-- cli@0.4.5 
  +-- esprima@4.0.0-dev 
  +-- peakle@0.0.1 
  +-- shelljs@0.1.4 
  `-- underscore@1.4.4 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-concat/package.json ]
+ npm install grunt-contrib-concat
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.1  extraneous
`-- grunt-contrib-concat@0.3.0 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-contrib-uglify/package.json ]
+ npm install grunt-contrib-uglify
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.1  extraneous
`-- grunt-contrib-uglify@0.2.7 
  +-- grunt-lib-contrib@0.6.1 
  | `-- zlib-browserify@0.0.1 
  `-- uglify-js@2.4.24 
    +-- async@0.2.10 
    +-- source-map@0.1.34 
    | `-- amdefine@1.0.1 
    +-- uglify-to-browserify@1.0.2 
    `-- yargs@3.5.4 
      +-- camelcase@1.2.1 
      +-- decamelize@1.2.0 
      +-- window-size@0.1.0 
      `-- wordwrap@0.0.2 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
+ [ -f node_modules/grunt-lint-inline/package.json ]
+ npm install grunt-lint-inline
my-project-name@0.1.0 /root/project/util/grunt
+-- grunt-cli@1.4.1  extraneous
`-- grunt-lint-inline@0.1.1 
  `-- grunt-contrib-jshint@0.3.0 

npm WARN my-project-name@0.1.0 No repository field.
npm WARN my-project-name@0.1.0 No license field.
npm WARN deprecated nodeunit@0.7.4: you are strongly encouraged to use other testing options
WARN engine tap@15.0.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine which@2.0.2: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine foreground-child@2.0.0: wanted: {"node":">=8.0.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine libtap@1.1.1: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine istanbul-lib-processinfo@2.0.2: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine jackspeak@1.4.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tap-mocha-reporter@5.0.1: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine mkdirp@1.0.4: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tcompare@5.0.4: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine coveralls@3.1.0: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine chokidar@3.5.1: wanted: {"node":">= 8.10.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine nyc@15.1.0: wanted: {"node":">=8.9"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine glob-parent@5.1.2: wanted: {"node":">= 6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine braces@3.0.2: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine is-binary-path@2.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine anymatch@3.1.1: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine fsevents@2.3.2: wanted: {"node":"^8.16.0 || ^10.6.0 || >=11.0.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine readdirp@3.5.0: wanted: {"node":">=8.10.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine picomatch@2.2.2: wanted: {"node":">=8.6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine fill-range@7.0.1: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine to-regex-range@5.0.1: wanted: {"node":">=8.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine binary-extensions@2.2.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine request@2.88.2: wanted: {"node":">= 6"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
WARN engine har-validator@5.1.5: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
WARN engine punycode@2.1.1: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine cross-spawn@7.0.3: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine path-key@3.1.1: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine shebang-command@2.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine shebang-regex@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine make-dir@3.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine p-map@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine istanbul-lib-coverage@3.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine aggregate-error@3.1.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine clean-stack@2.2.0: wanted: {"node":">=6"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine indent-string@4.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine bind-obj-methods@3.0.0: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine stack-utils@2.0.3: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine tap-parser@10.1.0: wanted: {"node":">= 8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine minipass@3.1.3: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine async-hook-domain@2.0.2: wanted: {"node":">=10"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine escape-string-regexp@2.0.0: wanted: {"node":">=8"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm ERR! Linux 4.15.0-1092-aws
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
npm ERR! typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
npm ERR! typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
npm ERR! typeerror     at /usr/share/npm/lib/install/deps.js:457:7
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
npm ERR! typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
npm ERR! typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
npm ERR! typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /root/project/util/grunt/npm-debug.log
Running "jshint:all" (jshint) task
>> 25 files lint free.

Done, without errors.

CircleCI received exit code 0
```

--- a/util/grunt/npm_install_grunt
+++ b/util/grunt/npm_install_grunt
@@ -12,7 +12,11 @@ type npm
 
 echo "Using npm to install needed grunt libraries"
 
-for module in grunt-cli grunt grunt-contrib-jshint grunt-contrib-concat \
+# hardcode an older version of grunt-cli to work around micromatch
+# version incompatibility found in #2713
+npm install grunt-cli@1.4.1
+
+for module in grunt grunt-contrib-jshint grunt-contrib-concat \
               grunt-contrib-uglify grunt-lint-inline; do
   if [ -f node_modules/${module}/package.json ]; then
     echo "${module} already installed"

